### PR TITLE
Add vcpkg.json and related files for building on Windows

### LINKED ISSUE #13962: Declare `vcpkg` dependencies in `vcpkg.json`
### Problem

Under Windows, the recommended way of installing Minetest dependencies is `vcpkg`.

But `vcpkg` have CMake integration ([including working with feature sets](https://learn.microsoft.com/en-us/vcpkg/users/buildsystems/cmake-integration#vcpkg_manifest_features)) and support the use of `vcpkg.json` to declare dependencies, version contraints and feature sets , so that CMake can handle itself package installation, IDEs can check installed package versions, etc

- [`vcpkg` manifest mode with CMake](https://learn.microsoft.com/en-us/vcpkg/consume/manifest-mode?tabs=cmake%2Cbuild-MSBuild )
- [`vcpkg.json` reference](https://learn.microsoft.com/en-us/vcpkg/reference/vcpkg-json)
- [`vcpkg` tutorial](https://learn.microsoft.com/en-us/vcpkg/get_started/get-started)

### Solutions

Add `vcpkg.json` with dependencies and feature sets, modify `CMakeLists.txt` to enable integration of feature sets.

### Alternatives

None

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by mark-wiemer**
Abandoned in favor of #15789 as this was (accidentally) done from my master branch, making life difficult

---

Closes #13962

- Goal of the PR: Ease installation on Windows for new contributors
- How does the PR work?: Provides declaration files and rewritten docs so users can install without providing manual complex commands
- Does it resolve any reported issue?: Yes, #13962
- Does this relate to a goal in [the roadmap](https://github.com/minetest/minetest/blob/master/doc/direction.md)?: Unsure
- If not a bug fix, why is this PR needed? What usecases does it solve?: n/a

## Changes

- Add `vcpkg.json` listing all required dependencies
- Rewrite `doc/compiling/windows.md` for clarity
    - Use VS 2022 as default VS version in docs
    - Use bundled vcpkg path instead of manually-installed path in docs, as vcpkg is hard to install
    - Remove `vcpkg install` step as that was duplicated by `cmake -G`
    - Use `cmake --build build` instead of `cmake --build .` to keep workspace cleaner
    - Remove some docs about installing vcpkg at a specific place, as it is never used directly. Instead, it's used in CMake, which doesn't have issues with spaces in paths
    - Add docs about how to install VS with C++ tools and how to confirm CMake is working
    - Add docs about how to run Luanti once built to confirm build process is complete
- Add disclaimer about "modders and game devs don't need to compile Luanti", instead directing them to the Luanti Modding Book. (happy to remove this if it's too much scope creep)
- Sort shell.nix dependency list for clarity

## How to test

Try following the new instructions on Windows and let me know if there are further points of clarification needed

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -138,3 +138,7 @@ lib/irrlichtmt
 
 # Generated mod storage database
 client/mod_storage.sqlite
+
+# mark-wiemer build stuff
+minetest-build
+vcpkg_installed
```
Comment (mark-wiemer) [inline .gitignore:144]:
```suggestion
```

File: doc/compiling/windows.md [hunk 2]
```diff
@@ -2,67 +2,82 @@
...
-
+-   [Visual Studio 2015 or newer](https://visualstudio.microsoft.com)
+-   [CMake](https://cmake.org/download/)
+-   [vcpkg](https://github.com/Microsoft/vcpkg)
+-   [Git](https://git-scm.com/downloads)
 
 ## Compiling and installing the dependencies
 
-It is highly recommended to use vcpkg as package manager.
+It is highly recommended to use vcpkg as package manager, installed in a location without spaces in the path for compatibility with CMake.
```
Comment (JosiahWI) [inline doc/compiling/windows.md:12]:
```suggestion
It is highly recommended to use vcpkg to manage dependencies, installed at a path without spaces for compatibility with the Luanti build configuration.
```

Comment (JosiahWI) [reply doc/compiling/windows.md:12]: CMake handles spaces fine. It's our fault spaces don't work, although anybody using spaces should know better than to test us.

File: doc/compiling/windows.md [hunk 3]
```diff
@@ -2,67 +2,82 @@
...
+
+```
+"C:\Program Filesvcpkg\vcpkg.exe" # bad, may have issues with CMake
```
Comment (JosiahWI) [inline doc/compiling/windows.md:15]:
```suggestion
"C:\Program Filesvcpkg\vcpkg.exe" # bad, may have issues
```

File: doc/compiling/windows.md [hunk 4]
```diff
@@ -2,67 +2,82 @@
...
+"C:\vcpkg\vcpkg.exe" # good, no spaces means fewer problems
+```
+
+If spaces are present, you may see errors like:
+
+```
+libtool:   error: 'Files/vcpkg/buildtrees/libiconv/x64-windows-dbg/lib/libcharset.la' is not a directory
+```
```
Comment (JosiahWI) [inline doc/compiling/windows.md:23]: I like the example. Great documentation.

File: doc/compiling/windows.md [hunk 5]
```diff
@@ -2,67 +2,82 @@
...
 
-- `curl` is optional, but required to read the serverlist, `curl[winssl]` is required to use the content store.
-- `openal-soft`, `libvorbis` and `libogg` are optional, but required to use sound.
-- `luajit` is optional, it replaces the integrated Lua interpreter with a faster just-in-time interpreter.
-- `gmp` and `jsoncpp` are optional, otherwise the bundled versions will be compiled
-- `gettext` is optional, but required to use translations.
+This command takes about 10-30 minutes to complete, depending on your device.
 
-There are other optional libraries, but they are not tested if they can build and link correctly.
+-   `curl` is optional, but required to read the serverlist, `curl[winssl]` is required to use the content store.
```
Comment (JosiahWI) [inline doc/compiling/windows.md:32]:
```suggestion
-   `curl` is optional, but required to read the serverlist; `curl[winssl]` is required to use the content store.
```

This should be a semicolon. If we're fixing the docs, let's make them shine.

**Review by JosiahWI — CHANGES_REQUESTED** [review]
I only started reviewing. I hope to review more, but I'll share what I have.

File: doc/compiling/windows.md (hunk 2 above)
Comment (LoneWolfHT) [reply doc/compiling/windows.md:12]: "path without spaces *in it*" sounds a little better to me

File: doc/compiling/windows.md (hunk 5 above)
Comment (appgurueu) [reply doc/compiling/windows.md:32]: `curl` is strongly recommended. It's required for anything HTTP(S). The serverlist or content browser (or the built-in update checker) are just some "mainmenu" examples of this; the Lua HTTP API also needs curl to work.

(Frankly we might consider making curl mandatory in the future?)

**Comment by LoneWolfHT** [comment]
I think this will also allow updating the CI that compiles the Visual Studio builds. I'll try to take a look after this is merged

File: doc/compiling/windows.md [hunk 6]
```diff
@@ -2,67 +2,82 @@
...
 
-### b) Using the vcpkg toolchain and the commandline
+### b) CLI
 
 Run the following script in PowerShell:
 
 ```powershell
 cmake . -G"Visual Studio 16 2019" -DCMAKE_TOOLCHAIN_FILE=D:/vcpkg/scripts/buildsystems/vcpkg.cmake -DCMAKE_BUILD_TYPE=Release -DENABLE_CURSES=OFF
 cmake --build . --config Release
 ```
```
Comment (JosiahWI) [inline doc/compiling/windows.md:71]: We should not be encouraging in-tree builds. Also, I think CMake should automatically pick the correct generator if it is left unspecified.

File: vcpkg.json [hunk 7]
```diff
@@ -0,0 +1,18 @@
+{
+  "dependencies": [
```
Comment (JosiahWI) [inline vcpkg.json:2]: These would be easier to review if they were lexicographically sorted.

Comment (JosiahWI) [reply vcpkg.json:2]: SDL2 is missing.

Comment (JosiahWI) [reply vcpkg.json:2]: Is libpng missing, or is that no longer a dependency?

File: doc/compiling/windows.md (hunk 2 above)
Comment (mark-wiemer) [reply doc/compiling/windows.md:12]:
```suggestion
It is highly recommended to use vcpkg to manage dependencies, installed at a path without spaces in it for compatibility with the Luanti build configuration.
```

File: doc/compiling/windows.md [hunk 8]
```diff
@@ -2,67 +2,82 @@
...
+```
+
+If spaces are present, you may see errors like:
+
+```
+libtool:   error: 'Files/vcpkg/buildtrees/libiconv/x64-windows-dbg/lib/libcharset.la' is not a directory
+```
 
 After you successfully built vcpkg you can easily install the required libraries:
 ```powershell
```
Comment (JosiahWI) [inline doc/compiling/windows.md:26]: CMake will also do installation when you invoke it, so I believe this step is technically unnecessary.

File: doc/compiling/windows.md (hunk 5 above)
Comment (mark-wiemer) [reply doc/compiling/windows.md:32]:
```suggestion
-   `curl` is highly recommended, as it's required to read the serverlist; `curl[winssl]` is required to use the content store.
```

**Review by JosiahWI — CHANGES_REQUESTED** [review]

File: doc/compiling/windows.md (hunk 6 above)
Comment (mark-wiemer) [reply doc/compiling/windows.md:71]: Hmm, personally I don't understand this, happy to take action if there are any suggestions :)

File: doc/compiling/windows.md (hunk 8 above)
Comment (mark-wiemer) [reply doc/compiling/windows.md:26]: Could you clarify what "this step" is? Maybe it can be removed in a separate PR? I'm personally re-building Luanti on a new machine now, I can try doing it without "this step", once I know what it is ;)

File: vcpkg.json (hunk 7 above)
Comment (mark-wiemer) [reply vcpkg.json:2]: This update was from October when I first built Luanti. I was able to run devtest with this configuration, but I'm happy to add requirements if they've changed since October.

Comment (mark-wiemer) [reply vcpkg.json:2]: Sorted the lines :)

File: doc/compiling/windows.md (hunk 6 above)
Comment (JosiahWI) [reply doc/compiling/windows.md:71]: Running `cmake` from the build directory will cause CMake to output its build artifacts in the source tree. It's a pain to clean up, and it easily clutters output from tooling. It's better to run `cmake -B <name_of_build_directory>` which will limit the scope of CMake's shenanigans to single directory, which can be deleted in its entirely if it gets messed up later to start fresh. It also makes it easy to maintain multiple builds with different configurations.

Comment (JosiahWI) [reply doc/compiling/windows.md:71]: I would suggest to change it to
```
cmake -B build -G"Visual Studio 16 2019" -DCMAKE_TOOLCHAIN_FILE=D:/vcpkg/scripts/buildsystems/vcpkg.cmake -DCMAKE_BUILD_TYPE=Release -DENABLE_CURSES=OFF
cmake --build build--config Release
```

I don't know how risky it would be to omit the `-G` option. It works on my machine because I use the CMake installed with the Visual Studio developer terminal, but I feel there's a risk that a different CMake installation might pick the wrong generator. I'm not well enough informed.

File: doc/compiling/windows.md (hunk 8 above)
Comment (JosiahWI) [reply doc/compiling/windows.md:26]: Running the vcpkg command to install dependencies. CMake will run it for you when you do a configuration.

File: vcpkg.json (hunk 7 above)
Comment (JosiahWI) [reply vcpkg.json:2]: I have both `sdl2` and `libpng` in my own vcpkg.json. I recall having to add `sdl2` in order to get Luanti to compile. It was the case that `libpng` was a dependency for Irrlicht, and I assume that is still the case. It may be that one of the other dependencies installs it transitively.

File: doc/compiling/windows.md (hunk 8 above)
Comment (JosiahWI) [reply doc/compiling/windows.md:26]: I think it's fine to leave it here. It doesn't cause any issues. It's just something to keep in mind if you weren't aware of it, since I think fewer steps here likely means less that can go wrong.

File: doc/compiling/windows.md (hunk 6 above)
Comment (mark-wiemer) [reply doc/compiling/windows.md:71]: I think we should keep the -G option for consistency

Comment (mark-wiemer) [reply doc/compiling/windows.md:71]: However, users will need to update the value based on their installed version of VS, ref #15336

File: vcpkg.json (hunk 7 above)
Comment (mark-wiemer) [reply vcpkg.json:2]: Ref #15336, it is a "works on my machine" solution, I'll keep investigating

**Comment by mark-wiemer** [comment]
Moving this to a draft as I work on some final tweaks, should re-publish later today :)

File: doc/compiling/windows.md [hunk 9]
```diff
@@ -2,67 +2,88 @@
...
-cmake . -G"Visual Studio 16 2019" -DCMAKE_TOOLCHAIN_FILE=D:/vcpkg/scripts/buildsystems/vcpkg.cmake -DCMAKE_BUILD_TYPE=Release -DENABLE_CURSES=OFF
+vs="Visual Studio 16 2019" # or "Visual Studio 17 2022", whatever matches your system
+toolchain_file="C:/vcpkg/scripts/buildsystems/vcpkg.cmake" # ensure this path is correct
+cmake . -G $vs -DCMAKE_TOOLCHAIN_FILE=$toolchain_file -DCMAKE_BUILD_TYPE=Release -DENABLE_CURSES=OFF
 cmake --build . --config Release
 ```
-Make sure that the right compiler is selected and the path to the vcpkg toolchain is correct.
-
 
 ## Windows Installer using WiX Toolset
```
Comment (mark-wiemer) [inline doc/compiling/windows.md:81]: Any advice on what to do with this "Windows Installer" section? I'm not sure what role it plays, or when it should come in. It mentions "compil[ing] binaries as above", but I don't remember seeing any "RUN_IN_PLACE" option

File: vcpkg.json (hunk 7 above)
Comment (mark-wiemer) [reply vcpkg.json:2]: Added SDL2, libpng wasn't necessary though.

Comment (mark-wiemer) [reply vcpkg.json:2]: Resolving as the original command was `vcpkg install zlib zstd curl[winssl] openal-soft libvorbis libogg libjpeg-turbo sqlite3 freetype luajit gmp jsoncpp gettext[tools] sdl2 --triplet x64-windows`, which doesn't include `libpng`

File: doc/compiling/windows.md (hunk 6 above)
Comment (mark-wiemer) [reply doc/compiling/windows.md]: Changed `.` to `build` :)

File: doc/compiling/windows.md [hunk 10]
```diff
@@ -1,68 +1,88 @@
...
 
-After you successfully built vcpkg you can easily install the required libraries:
-```powershell
-vcpkg install zlib zstd curl[winssl] openal-soft libvorbis libogg libjpeg-turbo sqlite3 freetype luajit gmp jsoncpp gettext[tools] sdl2 --triplet x64-windows
-```
+1. Confirm the installation.
+
+This will install the C++ compiler used in later steps. VS is also the recommended IDE for Luanti.
+
+You may notice that the C++ tools include CMake, but is should also be installed separately for compatibility with Luanti.
```
Comment (mark-wiemer) [inline doc/compiling/windows.md:23]:
```suggestion
You may notice that the C++ tools include CMake, but it should also be installed separately for compatibility with Luanti.
```

File: doc/compiling/windows.md [hunk 11]
```diff
@@ -1,68 +1,88 @@
...
+### Install dependencies
 
+1. Start up the CMake GUI (Win > search "cmake-gui" > open)
+1. Select **Browse Source...** and select `path/to/minetest` (where you've cloned the repo)
+1. Select **Browse Build...** and select `path/to/minetest/build` (a new folder that CMake will prompt to create)
+1. Select **Configure**
+1. Choose the right Visual Studio version and target platform. Currently, Luanti uses Visual Studio 16 2019, but newer VS versions should work as well. The VS version has to match the version of the installed dependencies.
+1. Choose **Specify toolchain file for cross-compiling**
+1. Click **Next**
+1. Select the vcpkg toolchain file, e.g. `C:/Program Files/Microsoft Visual Studio/2022/Community/VC/vcpkg/scripts/buildsystems/vcpkg.cmake`. Save this value for later.
```
Comment (mark-wiemer) [inline doc/compiling/windows.md:52]:
```suggestion
1. Select the vcpkg toolchain file, e.g. `C:/Program Files/Microsoft Visual Studio/2022/Community/VC/vcpkg/scripts/buildsystems/vcpkg.cmake`.
```

File: doc/compiling/windows.md [hunk 12]
```diff
@@ -1,68 +1,88 @@
...
+There are two ways to compile Luanti: via Visual Studio or via CLI.
+
+### Compile in Visual Studio
+
+14. Compile Luanti inside Visual Studio.
+    -   If you get "Unable to start program '...\x64\Debug\ALL_BUILD'. Access is denied", try compiling via the CLI instead.
+
+### Compile via CLI
+
+While in the `path/to/minetest` folder, run the following command:
```
Comment (JosiahWI) [inline doc/compiling/windows.md:70]: Maybe these sections should also mention that the Install Dependencies step must be completed before these will work.

Comment (mark-wiemer) [reply doc/compiling/windows.md:70]: Hmm, I figured folks would go top-down, would numbering the headings help?

File: doc/compiling/windows.md [hunk 13]
```diff
@@ -1,68 +1,88 @@
...
 
-After you successfully built vcpkg you can easily install the required libraries:
-```powershell
-vcpkg install zlib zstd curl[winssl] openal-soft libvorbis libogg libjpeg-turbo sqlite3 freetype luajit gmp jsoncpp gettext[tools] sdl2 --triplet x64-windows
-```
+1. Confirm the installation.
+
+This will install the C++ compiler used in later steps. VS is also the recommended IDE for Luanti.
+
+You may notice that the C++ tools include CMake, but it should also be installed separately for compatibility with Luanti.
```
[the remark this answers is not in the document]
Comment (mark-wiemer) [reply doc/compiling/windows.md:23]: cmake-gui wasn't included in my Start menu by default, I bet I can find it though, let me see if I can simplify the docs accordingly

File: doc/compiling/windows.md (hunk 12 above)
Comment (JosiahWI) [reply doc/compiling/windows.md:70]: Maybe it's clear enough. I'm just bringing up the discussion point for consideration, since I can imagine a scenario where someone skips to the part that says "Compile".

**Comment by JosiahWI** [comment]
@mark-wiemer The vcpkg.json works with Visual Studio 2022 and my CMake user preset, but I ran the configuration from the command line instead of using the CMake GUI. I do not use any GUI tools for my development, so I will need a little more time to walk through each step of the process outlined in the docs.

TL;DR vcpkg.json verified to be functionally correct on VS 2022, documentation still untested.

File: doc/compiling/windows.md (hunk 13 above)
[the remark this answers is not in the document]
Comment (mark-wiemer) [reply doc/compiling/windows.md]: Updated with instructions on how to add CMake CLI from VS to path and use that instead of GUI :)

File: doc/compiling/windows.md (hunk 12 above)
Comment (mark-wiemer) [reply doc/compiling/windows.md]: Numbered the headings and added a lead to each one clarifying that the previous step is required

**Review by JosiahWI — COMMENTED** [review]
I really appreciate this, thank you for improving these docs. I will try the GUI instructions if I have time - all the CLI instructions look great.

File: doc/compiling/windows.md (hunk 9 above)
Comment (JosiahWI) [reply doc/compiling/windows.md]: The `RUN_IN_PLACE` cache variable is used by Luanti to determine what paths to use for resources such as worlds and mods. I think I used the installer before, but it's been a while. We might have to rediscover how this works if no one remembers.

Comment (mark-wiemer) [reply doc/compiling/windows.md]: Resolving for now as out-of-scope for this PR

File: vcpkg.json [hunk 14]
```diff
@@ -0,0 +1,19 @@
...
+    "freetype",
+    "gettext",
+    "gmp",
+    "jsoncpp",
+    "libjpeg-turbo",
+    "libogg",
+    "libvorbis",
+    "luajit",
+    "openal-soft",
+    "opengl-registry",
```
Comment (sfan5) [inline vcpkg.json:13]: I belive opengl-registry is not needed anymore

File: vcpkg-configuration.json [hunk 15]
```diff
@@ -0,0 +1,14 @@
...
+    "repository": "https://github.com/microsoft/vcpkg"
+  },
+  "registries": [
+    {
+      "kind": "artifact",
+      "location": "https://github.com/microsoft/vcpkg-ce-catalog/archive/refs/heads/main.zip",
+      "name": "microsoft"
+    }
+  ]
+}
```
Comment (sfan5) [inline vcpkg-configuration.json:14]: two questions:
1. why do we need this file at all? AFAIK when the user sets up vcpkg they already clone the registry
2. I'm not sure what "vcpkg-ce" is and it doesn't look like we need it

File: doc/compiling/windows.md [hunk 16]
```diff
@@ -1,68 +1,133 @@
...
+
+## 2. Install Luanti dependencies
+
+Once VS and CMake are ready to use, you can install Luanti's dependencies. Luanti requires several C++ packages to function, which are installed by CMake and compiled by a C++ compiler bundled with VS. CMake uses a tool called vcpkg behind the scenes, which you may see referenced elsewhere in this project.
+
+### 2.1. Optional dependencies
+
+Some optional dependencies are recommended for advanced development. You can add them to `vcpkg.json` to install them along with the required dependencies.
+
+-   `curl` is highly recommended, as it's required to read the serverlist; `curl[winssl]` is required to use the content store.
```
Comment (sfan5) [inline doc/compiling/windows.md:57]:
```suggestion
-   `curl[winssl]` is highly recommended, as it's required to use the server list and content store.
```

File: doc/compiling/macos.md [hunk 17]
```diff
@@ -1,5 +1,7 @@
 # Compiling on MacOS
 
+If you're just creating mods or games with Luanti, you do not need to compile Luanti. Instead, follow Ruben Wardy's [Luanti Modding Book](https://rubenwardy.gitlab.io/minetest_modding_book) to get started modding. Compiling Luanti is only required if you plan to modify the Luanti engine itself.
```
Comment (sfan5) [inline doc/compiling/macos.md:3]: IMO we don't need this notice pasted over every instructions, but this PR isn't the right place anyway.

File: doc/compiling/windows.md [hunk 18]
```diff
@@ -1,68 +1,133 @@
...
+
+This will install the C++ compiler used in later steps. VS is also the recommended IDE for Luanti.
+
+### 1.2. CMake
+
+The CMake GUI is a great tool to manage first-time setup of Luanti. It's not required, but recommended for anyone unfamiliar with CMake or its CLI. CMake itself comes bundled with Visual Studio, but the CMake GUI does not, and must be installed separately.
+
+Install from [cmake.org/download](https://cmake.org/download/). Once installed, you should be able to run `cmake-gui` from the start menu:
+
+![cmake-gui in Windows start menu shows app result](./assets/cmake-gui-search.png)
```
Comment (sfan5) [inline doc/compiling/windows.md:31]: I think we can assume the user knows how to use the start menu

File: doc/compiling/assets/path-bin.png
Comment (sfan5) [inline doc/compiling/assets/path-bin.png:1]: please run these through a png optimizer like optipng or oxipng

**Review by sfan5 — CHANGES_REQUESTED** [review]

File: vcpkg-configuration.json (hunk 15 above)
Comment (mark-wiemer) [reply vcpkg-configuration.json:14]: 1. We may not need it, I'll see if it's an autogenerated file that can be ignored. Note that the new docs recommend using the `vcpkg` that comes built-in with VS, and even then vcpkg is used only indirectly through CMake. This simplifies the workflow a good amount IMO as a new dev.

2. `vcpkg-ce-catalog` looks like the registry used for installing artifacts. If we do need this file, I'll see what happens when this `registries` array is set to empty.

File: vcpkg.json (hunk 14 above)
Comment (mark-wiemer) [reply vcpkg.json:13]: Sure, I'll try rebuilding without this one

File: doc/compiling/macos.md (hunk 17 above)
Comment (mark-wiemer) [reply doc/compiling/macos.md:3]: Sure, I'll remove it from this PR

File: vcpkg-configuration.json (hunk 15 above)
Comment (mark-wiemer) [reply vcpkg-configuration.json:14]: Ugh, ran out of disk space, gonna have to try again

**Comment by mark-wiemer** [comment]
Proof of build before I cleanup the docs a bit :)

image

File: doc/compiling/windows.md [hunk 19]
```diff
@@ -1,68 +1,129 @@
...
 
-It is highly recommended to use vcpkg as package manager.
+### 1.1. Visual Studio (VS) with C++
+
+1. Install VS from [visualstudio.microsoft.com](https://visualstudio.microsoft.com)
+1. In the VS installer, select "Desktop development with C++":
+
+    ![VS installer showing desktop development with C++ selected](assets/vs-installer.png)
+
+1. Confirm the installation.
```
Comment (AFCMS) [inline doc/compiling/windows.md:19]: Why are all items numbered as 1?  It seems to be displayed properly as 1,2,3 in GitHub markdown preview but still.

**Review by AFCMS — COMMENTED** [review]
With `oxipng --opt max --strip all --zopfli` I managed to get a 10% size reduction for the screenshots.

Here are the resulting files:

[images.zip](https://github.com/user-attachments/files/18547286/images.zip)

**Comment by mark-wiemer** [comment]
Thanks, I'll go ahead and run the same command locally, and I'll ask around to find a reviewer to finally get this closed too

**Comment by mark-wiemer** [comment]
Hey @SmallJoker, I merged the upstream master into my master, is that OK? If needed, I can rebase the PR, but I'm happy for this to just be squashed when merged

File: README.md [hunk 20]
```diff
@@ -119,6 +119,8 @@ Command-line options
 Compiling
 ---------
 
+If you're just creating mods or games with Luanti, you do not need to compile Luanti. Instead, follow Ruben Wardy's [Luanti Modding Book](https://rubenwardy.gitlab.io/minetest_modding_book) to get started modding. Compiling Luanti is only required if you plan to modify the Luanti engine itself.
```
Comment (sfan5) [inline README.md:122]: this feels out of place here

File: doc/compiling/windows.md [hunk 21]
```diff
@@ -1,69 +1,129 @@
 # Compiling on Windows using MSVC
 
-## Requirements
+In this guide, we use `path/to/minetest` to refer to the repository. Minetest was the old name of Luanti, and as of this writing, the repo has not been renamed.
```
Comment (sfan5) [inline doc/compiling/windows.md:3]: not correct anymore :)

File: doc/compiling/windows.md (hunk 19 above)
Comment (sfan5) [reply doc/compiling/windows.md:19]: markdown has automatic numbering

File: doc/compiling/windows.md [hunk 22]
```diff
@@ -1,69 +1,129 @@
...
+-   `openal-soft`, `libvorbis` and `libogg` are optional, but required to use sound.
+-   `luajit` is optional, it replaces the integrated Lua interpreter with a faster just-in-time interpreter.
+-   `gmp` and `jsoncpp` are optional, otherwise the bundled versions will be compiled
+-   `gettext` is optional, but required to use translations.
+
+### 2.2. Install dependencies
+
+You can install dependencies via CLI or via CMake GUI, whichever your prefer.
+
+To install dependencies via CLI, you can run this program in PowerShell:
```
Comment (sfan5) [inline doc/compiling/windows.md:63]:
```suggestion
To install dependencies via CLI, run the following commands in PowerShell:
```

File: doc/compiling/windows.md [hunk 23]
```diff
@@ -1,69 +1,129 @@
...
 
 ```powershell
-vcpkg install zlib zstd curl[winssl] openal-soft libvorbis libogg libjpeg-turbo sqlite3 freetype luajit gmp jsoncpp gettext[tools] opengl-registry --triplet x64-windows
+# Run this while in `path/to/luanti`
+$vs="Visual Studio 17 2022" # or "Visual Studio 16 2019", whatever matches your machine
+$toolchain_file="C:/Program Files/Microsoft Visual Studio/2022/Community/VC/vcpkg/scripts/buildsystems/vcpkg.cmake" # or wherever you installed vcpkg.cmake manually
+mkdir build
+cmake -S . -B build -G "$vs" -DCMAKE_TOOLCHAIN_FILE="$toolchain_file" -DCMAKE_BUILD_TYPE=Release -DENABLE_CURSES=OFF
```
Comment (sfan5) [inline doc/compiling/windows.md:70]: running a configure installs the dependencies? that is a bit confusing even if true.
any reason not to continue recommending the `vcpkg` call like before?

File: shell.nix [hunk 24]
```diff
@@ -6,20 +6,20 @@ pkgs.mkShell {
...
-    pkgs.openal
-    pkgs.curl
     pkgs.libvorbis
-    pkgs.libogg
-    pkgs.gettext
-    pkgs.freetype
+    pkgs.openal
     pkgs.sqlite
+    pkgs.zlib
+    pkgs.zstd
```
Comment (sfan5) [inline shell.nix:23]: what changed here and why?

File: vcpkg.json [hunk 25]
```diff
@@ -0,0 +1,18 @@
...
+    "freetype",
+    "gettext",
+    "gmp",
+    "jsoncpp",
+    "libjpeg-turbo",
+    "libogg",
+    "libvorbis",
+    "luajit",
+    "openal-soft",
+    "sdl2",
```
Comment (sfan5) [inline vcpkg.json:13]: this not correct as of the "revert sdl for 5.11" commit

File: vcpkg.json [hunk 26]
```diff
@@ -0,0 +1,18 @@
+{
+  "dependencies": [
+    "curl",
+    "freetype",
+    "gettext",
+    "gmp",
+    "jsoncpp",
+    "libjpeg-turbo",
+    "libogg",
+    "libvorbis",
```
Comment (sfan5) [inline vcpkg.json:10]: i think it would make sense to include all optional depends.
right now it's inconsistent (sound support, but no SSL)

**Review by sfan5 — CHANGES_REQUESTED** [review]
I think it would also make sense if the msvc CI build used the new instructions instead of manually installing stuff.
This also allows us to verify that they are correct.

**Comment by sfan5** [comment]
We can also consider moving the GUI instructions for developing using Visual Studio to dev.luanti.org so it doesn't have to go through the slow engine review process.

**Comment by mark-wiemer** [comment]
Thanks for all the comments. I'm worried my merge with master inadvertently introduced changes, I'll double-check this evening (about 12 hours from now)

**Comment by medleymedley** [comment]
As someone who almost gave up on setting up a build environment on Windows I'm in full support of this lol, it took some manual looking through compilation errors at all stages to get it up and running. 

Changing the requirement to Visual Studio Community 2022 would help a lot, Microsoft gatekeeps the 2019 version and just installing its component didn't work for me either. The project compiles fine on 2022, I just had to set it up more manally through GUI, as it seemed the CLI setup it requires 2019 .

The easier it is to set up, the easier it is for new devs to contribute and test their code, or to test on different platforms, so this is a valuable PR imo.

**Comment by mark-wiemer** [comment]
Yeah, I haven't forgotten about this one, maybe I'll check it out this weekend. Maybe ;)

File: shell.nix (hunk 24 above)
Comment (mark-wiemer) [reply shell.nix:23]: reordered the list to be alphabetical for easy manual scanning and comparison with vcpkg.json

File: README.md (hunk 20 above)
Comment (mark-wiemer) [reply README.md:122]:
```suggestion
```

File: doc/compiling/windows.md (hunk 19 above)
Comment (mark-wiemer) [reply doc/compiling/windows.md:19]: Yes, I use auto-numbering as it's easier to move or insert items later. Resolving

File: doc/compiling/windows.md (hunk 23 above)
Comment (mark-wiemer) [reply doc/compiling/windows.md:70]: The `vcpkg` call is unnecessary and overcomplicates the build process, in short.

File: vcpkg.json (hunk 26 above)
Comment (mark-wiemer) [reply vcpkg.json:10]: Makes sense, let me update this list and rebuild

**Comment by mark-wiemer** [comment]
Working on removing merge commits and cleaning up my master branch

File: shell.nix (hunk 24 above)
Comment (mark-wiemer) [reply shell.nix:23]: Undoing this change as I'm no longer using NixOS and I don't want to push untested changes

**Comment by mark-wiemer** [comment]
Abandoning in favor of #15789
